src/Tests/Framework/Integration/Process/ElasticsearchNode.cs

```diff
@@ -128,89 +145,131 @@ string prefix
-						return Observable.Empty<ElasticsearchMessage>();
-					}
-				}
-			}
-			var settings = new string[]
-			{
-				$"-Des.cluster.name={this.ClusterName}",
-				$"-Des.node.name={this.NodeName}",
-				$"-Des.path.repo={this.RepositoryPath}",
-				$"-Des.script.inline=on",
-				$"-Des.script.indexed=on",
-				$"--node.testingcluster true"
-			}
-			.Concat(additionalSettings ?? Enumerable.Empty<string>())
-			.OrderByDescending(s =>
-				s.StartsWith("-d", StringComparison.OrdinalIgnoreCase) ||
-				s.StartsWith("-p", StringComparison.OrdinalIgnoreCase))
-			.ThenBy(s => s);
+			var alreadyRunning = UseAlreadyRunningInstance(handle);
+			if (alreadyRunning != null) return alreadyRunning;
 
 			this._process = new ObservableProcess(this.Binary, settings.ToArray());
 
 			var observable = Observable.Using(() => this._process, process => process.Start())
 				.Select(consoleLine => new ElasticsearchMessage(consoleLine));
 			this._processListener = observable.Subscribe(onNext: s => HandleConsoleMessage(s, handle));
 
-			if (!handle.WaitOne(timeout, true))
-			{
-				this.Stop();
-				throw new Exception($"Could not start elasticsearch within {timeout}");
-			}
-
-			return observable;
-		}
+			if (handle.WaitOne(this.HandleTimeout, true)) return observable;
 
+			this.Stop();
+			throw new Exception($"Could not start elasticsearch within {this.HandleTimeout}");
+		}
+
+#if DOTNETCORE
+		private IObservable<ElasticsearchMessage> UseAlreadyRunningInstance(Signal handle)
+#else
+		private IObservable<ElasticsearchMessage> UseAlreadyRunningInstance(ManualResetEvent handle)
+#endif
+		{
+			if (!_doNotSpawnIfAlreadyRunning) return null;
+
+			var client = TestClient.GetClient();
+			var alreadyUp = client.RootNodeInfo();
+
+			if (!alreadyUp.IsValid) return null;
+
+			var checkPlugins = client.CatPlugins();
+
+			var missingPlugins = SupportedPlugins.Keys.Except(checkPlugins.Records.Select(r => r.Component)).ToList();
+			if (missingPlugins.Any())
+				throw new Exception($"Already running elasticsearch missed the following plugin(s): {string.Join(", ", missingPlugins)}.");
+
+			this.Started = true;
+			this.Port = 9200;
+			this.Info = new ElasticsearchNodeInfo(alreadyUp.Version.Number, null, alreadyUp.Version.LuceneVersion);
+			this._blockingSubject.OnNext(handle);
+			if (!handle.WaitOne(this.HandleTimeout, true))
+				throw new Exception($"Could not launch tests on already running elasticsearch within {this.HandleTimeout}");
+
+			ValidateLicense();
+
+			return Observable.Empty<ElasticsearchMessage>();
+		}
+
+
+		private void ValidateLicense()
+		{
+			var client = TestClient.GetClient();
+			var license = client.GetLicense();
+			if (license.IsValid && license.License.Status == LicenseStatus.Active) return;
+
+			var exceptionMessageStart = "Server has license plugin installed, ";
+			var licenseFile = Environment.GetEnvironmentVariable("ES_LICENSE_FILE", EnvironmentVariableTarget.Machine);
+			if (!string.IsNullOrWhiteSpace(licenseFile))

```

Comment:
Should we also check for existence of the license here and continue if it isn't there?